src/rust/sysmon-subgraph-generator/src/metrics.rs

```diff
@@ -0,0 +1,25 @@
+use failure::Error;
+use grapl_observe::metric_reporter::{MetricReporter, TagPair};
+use log::*;
+
+#[derive(Clone)]
+pub struct SysmonSubgraphGeneratorMetrics {
+    pub metric_reporter: MetricReporter,
+}
+
+impl SysmonSubgraphGeneratorMetrics {
+    pub fn report_handle_event_success(&mut self, failed: &Option<Error>) {
+        let reported_status = if let Some(ref e) = failed {
```

Comment:
would this be better off as a `failed.map_or_else("completed", "failed")`?

Reply:
At minimum instead of `ref e` it should just be `_`. map_or_else would be fine too, no real preference from my end, both are fairly clear.